mqtt/edgelet-client/src/workload/client.rs

```diff
@@ -86,9 +86,11 @@ impl WorkloadClient {
             .map_err(|e| ApiError::ReadResponse(Box::new(e)))?;
 
         if status != StatusCode::CREATED {
-            let text =
-                str::from_utf8(body.bytes()).map_err(|e| ApiError::ReadResponse(Box::new(e)))?;
-            return Err(ApiError::UnsuccessfulResponse(status, text.into()).into());
+            let mut text = String::new();
+            body.reader()
+                .read_to_string(&mut text)
+                .map_err(|e| ApiError::ReadResponse(Box::new(e)))?;
+            return Err(ApiError::UnsuccessfulResponse(status, text).into());
         }
```

Comment:
Are these blocks all the same? Should we make a smaller helper? If we end up upgrading again later then maybe it will be easier?

Reply:
I'm not sure I understood suggestion ...